### PR TITLE
Converting to CoffeeScript

### PR DESCRIPTION
**Pull request by emmetcampion**

### CODE REVIEW

**Comment by ripexz** [comment]
:+1:

File: package.json [hunk 1]
```diff
@@ -12,12 +18,28 @@
...
+    "gulp-mocha": "^0.4.1",
+    "gulp-util": "^2.2.20",
+    "teamwork-coffeelint-rules": "^0.0.1"
   },
   "dependencies": {},
-  "devDependencies": {},
-  "main": "index.js"
-}
\ No newline at end of file
+  "publishConfig": {
```
Comment (adam-lynch) [inline package.json]: This shouldn't be there. This is only for private modules.

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Logs
```
Comment (adam-lynch) [inline .npmignore]: I might need to update the slush generator but it's better if our `.npmignore`s ignore everything then unignore anything necessary. Like https://github.com/Teamwork/chat-renderer/blob/master/.npmignore

**Comment by emmetcampion** [comment]
Thanks Adam them changes are made now.

**Comment by 4ver** [comment]
Savage!

File: src/index.coffee [hunk 3]
```diff
@@ -0,0 +1,82 @@
+exec = require('child_process').exec
+os = require('os')
+path = require('path')
+
+class idle
```
Comment (adam-lynch) [inline src/index.coffee]: Uppercase :+1:

File: package.json [hunk 4]
```diff
@@ -12,12 +18,25 @@
...
+  "homepage": "https://github.com/teamwork/node-afk",
+  "devDependencies": {
+    "chai": "^1.9.2",
+    "coffee-script": "^1.9.2",
+    "gulp": "^3.8.11",
+    "gulp-coffee": "^2.0.1",
+    "gulp-coffeelint": "^0.3.4",
+    "gulp-coffeelint-threshold": "^1.0.0",
+    "gulp-load-plugins": "^0.5.3",
+    "gulp-mocha": "^0.4.1",
```
Comment (adam-lynch) [inline package.json]: Not needed. Only for tests.

File: package.json [hunk 5]
```diff
@@ -12,12 +18,25 @@
...
+  "author": "Emmet Campion <Emmet@teamwork.com>",
+  "contributors": [
+    "Donal Linehan <me@don.al>"
+  ],
+  "bugs": {
+    "url": "https://github.com/teamwork/node-afk/issues"
+  },
+  "homepage": "https://github.com/teamwork/node-afk",
+  "devDependencies": {
+    "chai": "^1.9.2",
```
Comment (adam-lynch) [inline package.json]: Same for this; not needed.

File: src/index.coffee [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    # counts how long user is afk in seconds
+    # :callback - {function}
+    tick: (callback) ->
+        callback ?= ->
+
+        if /^win/.test process.platform
+            cmd = '"' + path.join(__dirname, 'bin', 'idle.exe') + '"'
+            exec cmd, (error, stdout, stderr) ->
+                if error
+                    calback(0, error)
```
Comment (bakaat) [inline src/index.coffee]: fix me, and line 90 and 98

**Pull request closed without merging**